### ISSUE TITLE
Non-Steam games no longer work correctly

### ISSUE THREAD

**Issue by pedrovay2003** [ValveSoftware/steam-for-linux#10891]
#### Your system information

* SteamOS version: May 13th
* Opted into Steam client beta?: No
* Opted into SteamOS beta?: Yes and No
* Have you checked for updates in Settings > System?: Yes

#### Please describe your issue in as much detail as possible:

As of the latest Steam Deck stable update, non-Steam games added in Desktop Mode don't appear in Gaming Mode; returning to Desktop Mode, they're there. Non-Steam games that I had added before the update are still there in Gaming Mode, but they don't appear in Desktop Mode anymore. I tried the Beta branch, but it's the same behavior there, and another user reported the same behavior in the Preview branch, as well.

As I was typing this, the Steam Windows client updated, and all of my non-Steam games were removed from my library.

#### Steps for reproducing this issue:

1. Add a non-Steam game to Steam in Desktop Mode
2. Return to Gaming Mode, where that non-Steam game will be missing
3. Check Desktop Mode for any non-Steam games added prior to this update, and they will be missing in Desktop Mode, even though they appear in Gaming Mode

**Comment by kisak-valve** [comment]
Hello @pedrovay2003, you've described an issue with the Steam client instead of SteamOS as a platform, so I've transferred this issue report to the steam-for-linux issue tracker.

**Comment by midnightcatwalk** [comment]
Running into this as well on Deck (non-Steam games not populating in Steam in either Game or Desktop mode). 

The non-Steam games are still present in the shortcuts.vdf file, and shutting down and restarting the Deck only bery occasionally 2/10 tries for me) restores them to Steam.

**Comment by Trebolt23** [comment]
Can confirm I am having the same issue, annoyed that it's a thing at all but at least I know I'm not alone/it wasn't something I did.

**Comment by italoghost** [comment]
I am experiencing this at my desktop. Steam updated itself and every non-steam game has vanished.

**Comment by rKsanu2MMYvypWePtQWM** [comment]
same

**Comment by Bizzo499** [comment]
Hello, I'm also facing the same issue.
On my Deck, shortcuts that I added before the update are still in gaming mode while in the desktop client, all of them are missing.
If I try to add new ones, they will not appear in gaming mode.
I tried to switch to the beta channel but nothing changed.

**Comment by midnightcatwalk** [comment]
From the Steam Deck Discord:

Still no fix from what I've been researching, but on the Steam Deck discord a user named "nadd" figured out a temporary fix that worked for me. This was his post:

"After the last Steam Client update I was greeted by empty collections, everything launched without much issue.

The real issue that none of said nonsteam games I saw in game mode are present in desktop mode, literally not a single one. Regular steam games are there.

It's like the game mode library and desktop one are using completely different configs.

On top of that, when I add a new nonsteam game in desktop mode, I can run it fine in desktop, but I can't find it anywhere in game mode. If I add it to favorites or a new collection in desktop mode, a new category gets created in game mode, but it's completely empty: favorites included.

**This has me completely stumped.desktop mode steam client now uses /home/deck/.local/share/Steam/userdata/0/config/shortcuts.vdf

while gamemode uses /home/deck/.local/share/Steam/userdata/<userID>/config/shortcuts.vdf

from what I remember before the last update the desktop steam client shared the userID folder, when it came to the shortcuts.vdf files anyway"

Quick fix for this was to symlink the shortcuts.vdf from one folder to another, seems to mostly work though for how long I don't know, Valve needs to step up here."**

Hope that helps.

**Comment by italoghost** [comment]
@midnightcatwalk It worked!

**Comment by pedrovay2003** [comment]
I have an update to this as far as the desktop Steam client goes. This comes from a user on the official Steam forums: 

"Adding the shortcuts.vdf file to Steam\userdata\0\config can bring them back. But it's so weird because, if you delete the vdf file, sometimes adding a non-steam shortcut then restarting Steam adds a file to Steam\userdata\0\config and other times it adds it to userdata\<USERID>\config. It's like Steam alternates between where the file is supposed to be. And sometimes Steam deletes the file in the <USERID> folder upon restart."

**Comment by pedrovay2003** [comment]
I have an update to this as far as the desktop Steam client goes. This comes from a user on the official Steam forums: 

"Adding the shortcuts.vdf file to Steam\userdata\0\config can bring them back. But it's so weird because, if you delete the vdf file, sometimes adding a non-steam shortcut then restarting Steam adds a file to Steam\userdata\0\config and other times it adds it to userdata\<USERID>\config. It's like Steam alternates between where the file is supposed to be. And sometimes Steam deletes the file in the <USERID> folder upon restart."

**Comment by Devinerz** [comment]
In Game Mode, go to System > Updates and switch to Preview mode (from Stable or Beta) and install/reboot/let update finish. 
They've made a fix for this as of May 15th I think.

My games weren't visible at first, so I had to re-add them via desktop mode but they work fine after that.

**Comment by midnightcatwalk** [comment]
> In Game Mode, go to System > Updates and switch to Preview mode (from Stable or Beta) and install/reboot/let update finish. They've made a fix for this as of May 15th I think.
> 
> My games weren't visible at first, so I had to re-add them via desktop mode but they work fine after that.

Having to re-add everything doesn’t sound like a fix. But today’s stable client update does. Will see.

**Comment by pedrovay2003** [comment]
I updated my Deck to the latest Stable version, and it didn't seem to fix anything; I removed all of my non-Steam games and re-added them just to make sure that they were definitely there, and they only showed up in Desktop Mode, not in Gaming Mode. I had to mess with the shortcuts.vdf files mentioned above to get everything to show up in both modes.

**Comment by Devinerz** [comment]
> I updated my Deck to the latest Stable version, and it didn't seem to fix anything; I removed all of my non-Steam games and re-added them just to make sure that they were definitely there, and they only showed up in Desktop Mode, not in Gaming Mode. I had to mess with the shortcuts.vdf files mentioned above to get everything to show up in both modes.

Pretty sure Stable does not have this fix, so I'm not sure why you expected a fix? 
The latest version of Stable is the one where this is a problem.

If you read the link below, you'll see the "Steam Deck Client 5/16/24 Changelog" part. 
https://steamdeckhq.com/news/steam-deck-client-steamos-preview-neat-updates/

If you've tinkered with your shortcut files already, I'm not sure if that'll cause a problem. 

I've never had an issue viewing games in Desktop mode, only Game mode.
But I removed/re-added a few times, so maybe that's why I had to re-add my games once I switched to Preview mode.

**Comment by midnightcatwalk** [comment]
> > I updated my Deck to the latest Stable version, and it didn't seem to fix anything; I removed all of my non-Steam games and re-added them just to make sure that they were definitely there, and they only showed up in Desktop Mode, not in Gaming Mode. I had to mess with the shortcuts.vdf files mentioned above to get everything to show up in both modes.
> 
> Pretty sure Stable does not have this fix, so I'm not sure why you expected a fix? The latest version of Stable is the one where this is a problem.
> 
> If you read the link below, you'll see the "Steam Deck Client 5/16/24 Changelog" part. https://steamdeckhq.com/news/steam-deck-client-steamos-preview-neat-updates/
> 
> If you've tinkered with your shortcut files already, I'm not sure if that'll cause a problem.
> 
> I've never had an issue viewing games in Desktop mode, only Game mode. But I removed/re-added a few times, so maybe that's why I had to re-add my games once I switched to Preview mode.

Check the “May 17th” update:

https://www.steamdeck.com/en/news

**Comment by Devinerz** [comment]
> > > I updated my Deck to the latest Stable version, and it didn't seem to fix anything; I removed all of my non-Steam games and re-added them just to make sure that they were definitely there, and they only showed up in Desktop Mode, not in Gaming Mode. I had to mess with the shortcuts.vdf files mentioned above to get everything to show up in both modes.
> > 
> > 
> > Pretty sure Stable does not have this fix, so I'm not sure why you expected a fix? The latest version of Stable is the one where this is a problem.
> > If you read the link below, you'll see the "Steam Deck Client 5/16/24 Changelog" part. https://steamdeckhq.com/news/steam-deck-client-steamos-preview-neat-updates/
> > If you've tinkered with your shortcut files already, I'm not sure if that'll cause a problem.
> > I've never had an issue viewing games in Desktop mode, only Game mode. But I removed/re-added a few times, so maybe that's why I had to re-add my games once I switched to Preview mode.
> 
> Check the “May 17th” update:
> 
> https://www.steamdeck.com/en/news
> 
> Hasn’t hit my Deck yet, but sounds promising

Ahh, good spot, I hadn't seen that.
I wonder have they only partially promoted some of the Preview branches changes into Stable? The wording is fairly different between the two. 

Latest Stable update says:
Fixed a case where the non-steam game list could load from an invalid user profile.

Latest Preview update says:
Fixed non-steam games not showing in search results.
Fixed a bug that could result in non-primary steam library folders to be forgotten.

**Comment by midnightcatwalk** [comment]
> > Replying to [#10891 (comment)](https://github.com/ValveSoftware/steam-for-linux/issues/10891#issuecomment-2118672849)
> 
> Ahh, good spot, I hadn't seen that. I wonder have they only partially promoted some of the Preview branches changes into Stable? The wording is fairly different between the two.
> 
> **Latest Stable update says:** Fixed a case where the non-steam game list could load from an invalid user profile.
> 
> **Latest Preview update says:** Fixed non-steam games not showing in search results. Fixed a bug that could result in non-primary steam library folders to be forgotten.

Turns out that they mislabeled the May 17th client update as a Stable update, when in fact it’s Beta/Preview. So it’s presumably a different fix.

**Comment by lostgoat** [comment]
Correct, the fix for this issue is currently available in the Beta/Preview channels. It hasn’t landed in stable yet.

**Comment by ThisNekoGuy** [comment]
[I actually mentioned this](https://github.com/ValveSoftware/SteamOS/issues/1413) early this month on this 6th (about 3 weeks ago) over on the SteamOS tracker too
Was wondering if it was ever going to get looked into because I didn't really get a response...

**Comment by Thornskade** [comment]
So have all non-Steam games been deleted? What happens to the input configs? How cant his be prevented in the future?

**Comment by lostgoat** [comment]
They haven’t been deleted. There was a random chance they wouldn’t load on startup.

There is a fix in the steam client beta.

**Comment by Thornskade** [comment]
But I switched to bith the beta and preview several times and they never showed up. Except for a single non-Steam game which I had added after the bug, before I knew what was going on. I can only conclude they've been deleted

**Comment by midnightcatwalk** [comment]
> But I switched to both the beta and preview several times and they never showed up. Except for a single non-Steam game which I had added after the bug, before I knew what was going on. I can only conclude they've been deleted

But are you currently on the most recent beta/preview client? (not OS)

You could also check your shortcuts.vdf file and see if the non-Steam games are still there, and just not being read (the bug) or actually deleted.

**Comment by Thornskade** [comment]
> > But I switched to both the beta and preview several times and they never showed up. Except for a single non-Steam game which I had added after the bug, before I knew what was going on. I can only conclude they've been deleted
> 
> But are you currently on the most recent beta/preview client? (not OS)
> 
> You could also check your shortcuts.vdf file and see if the non-Steam games are still there, and just not being read (the bug) or actually deleted.

Yeah, according to FSearch there are multiple shortcuts.vdf files on my system. Only two of them have any contents, and both show the one game I added after the update introduced the bug

**Comment by Bultom** [comment]
Hello,

Bumping this issue cause the bug came back
Steam last stable or beta client (as of 23/10/24) , W11 last stable update
